src/Kdyby/ElasticSearch/Diagnostics/Panel.php

```diff
@@ -108,7 +108,13 @@ public function getPanel()
 				return !is_array($data) ? Json::decode($data, Json::FORCE_ARRAY) : $data;
 
 			} catch (Nette\Utils\JsonException $e) {
-				return $data;
+				try {
+					return array_map(function($row) {
+						return Json::decode($row, Json::FORCE_ARRAY);
+					}, explode("\n", $data));
```

Comment:
Well, if it isn't properly escaped then the whole request is invalid (or at least the parts from the newline onwards). ES defines that multi requests are separated using a **newline** (see the link in the PR description).

And in that case (which would probably mean a bug in json_encode or Elastica; both being out of the scope of the panel) the non-decoded string is printed (just like now).